packages/grid/_modules_/grid/hooks/features/useColumnResize.tsx

```diff
@@ -1,204 +1,130 @@
-    (newWidth: number) => {
-      logger.debug(`Updating width to ${newWidth} for col ${currentColDefRef.current!.field}`);
-      if (currentColDefRef.current) {
-        currentColDefRef.current.width = newWidth;
-      }
-      if (currentColElem.current) {
-        currentColElem.current.style.width = `${newWidth}px`;
-        currentColElem.current.style.minWidth = `${newWidth}px`;
-        currentColElem.current.style.maxWidth = `${newWidth}px`;
-      }
-      if (dataContainerElemRef.current) {
-        const diffWithPrev = newWidth - currentColPreviousWidth.current!;
-        dataContainerElemRef.current.style.minWidth = `${
-          dataContainerPreviousWidth.current! + diffWithPrev
-        }px`;
-
-        if (isLastColumn.current) {
-          apiRef.current.scroll({ left: dataContainerPreviousWidth.current! + diffWithPrev });
-        }
-      }
-      if (currentColCellsElems.current) {
-        currentColCellsElems.current.forEach((el) => {
-          const div = el as HTMLDivElement;
-          div.style.width = `${newWidth}px`;
-          div.style.minWidth = `${newWidth}px`;
-          div.style.maxWidth = `${newWidth}px`;
-        });
-      }
-    },
-    [apiRef, logger],
-  );
+    });
 
-  const handleMouseEnter = React.useCallback((): void => {
-    if (mouseLeftTimeout.current != null) {
-      clearTimeout(mouseLeftTimeout.current);
-    }
-  }, []);
-  const handleMouseLeave = React.useCallback((): void => {
-    if (
-      isLastColumn.current &&
-      resizingMouseMove.current &&
-      resizingMouseMove.current.y >= 0 &&
-      resizingMouseMove.current.y <= headerHeight &&
-      currentColDefRef.current
-    ) {
-      logger.debug(`Mouse left and same row, so extending last column width of 100`);
-
-      // we are resizing the last column outside the window
-      updateWidth(currentColDefRef.current.width! + 10);
-      mouseLeftTimeout.current = setTimeout(() => {
-        stopResize();
-      }, MOUSE_LEFT_TIMEOUT);
-    } else if (isResizing) {
-      mouseLeftTimeout.current = setTimeout(() => {
-        stopResize();
-      }, MOUSE_LEFT_TIMEOUT);
+    logger.debug(
+      `Updating col ${colDefRef.current!.field} with new width: ${colDefRef.current!.width}`,
+    );
+  });
+
+  const handleMouseDown = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+    // Only handle left clicks
+    if (event.button !== 0) {
+      return;
     }
-  }, [headerHeight, logger, stopResize, updateWidth]);
 
-  const handleMouseMove = React.useCallback(
-    (ev: MouseEvent): void => {
-      if (isResizing.current) {
-        const target = ev.currentTarget! as HTMLDivElement;
-        const rect = target.getBoundingClientRect();
+    // Avoid text selection
+    event.preventDefault();
 
-        resizingMouseMove.current = { x: ev.clientX - rect.left, y: ev.clientY - rect.top };
+    colElementRef.current = event.currentTarget.closest('.MuiDataGrid-colCell') as HTMLDivElement;
```

Comment:
What if we change the css classname?
It should use cssConstant. Also there are helper function for retrieving cells and doing dom manipulation

Reply:
If we change the class name we will need to do a search & replace in the whole codebase. I think that test should cover this case, not constants.

Reply:
search & replace => that's why we have constants

Reply:
In this case, I think that the constants help so we get a static analysis code failure if the value is wrong. The static analysis can run faster than a test. I personally think that a constant and a class name can search & replace as well in the codebase. However, if you get the class name wrong, you might get a fail in the end-to-end test but you will spend quite some time figure out why (if the class name was wrong). I will use a constant.